pkg/image/builder.go

```diff
@@ -209,41 +217,21 @@ func listImagesInFile(contents []byte, handler processImagesFunc) error {
 	return nil
 }
 
-func saveOneImage(srcRegistry registry.RegistryOptions, imagesDir string, image string, appSlug string, reportWriter io.Writer) error {
-	imageRef, err := imageRefImage(image)
-	if err != nil {
-		return errors.Wrap(err, "failed to parse image ref")
-	}
-
-	imageFormat := "docker-archive"
-	pathInBundle := imageRef.pathInBundle(imageFormat)
-	archiveName := filepath.Join(imagesDir, pathInBundle)
-	destDir := filepath.Dir(archiveName)
-
-	if err := os.MkdirAll(destDir, 0755); err != nil {
-		return errors.Wrap(err, "failed to create destination dir")
-	}
-
+func copyOneImage(srcRegistry, destRegistry registry.RegistryOptions, image string, appSlug string, reportWriter io.Writer) ([]kustomizeimage.Image, error) {
 	policy, err := signature.NewPolicyFromBytes(imagePolicy)
 	if err != nil {
-		return errors.Wrap(err, "failed to read default policy")
+		return nil, errors.Wrap(err, "failed to read default policy")
 	}
 	policyContext, err := signature.NewPolicyContext(policy)
 	if err != nil {
-		return errors.Wrap(err, "failed to create policy")
-	}
-
-	destStr := fmt.Sprintf("%s:%s", imageFormat, archiveName)
-	destRef, err := alltransports.ParseImageName(destStr)
-	if err != nil {
-		return errors.Wrapf(err, "failed to parse local image name: %s", destStr)
+		return nil, errors.Wrap(err, "failed to create policy")
 	}
 
 	sourceCtx := &types.SystemContext{}
 
 	isPrivate, err := isPrivateImage(image)
 	if err != nil {
-		return errors.Wrapf(err, "failed to check if image is private: %s", image)
+		return nil, errors.Wrapf(err, "failed to check if image is private: %s", image)
```

Comment:
i see the name is here. but its a bit easier to just add it in the wrapping function and remove from everything in this function